tools/cpp/CROSSTOOL.tpl

```diff
@@ -990,12 +978,140 @@ toolchain {
       action: 'c-compile'
       action: 'c++-compile'
       flag_group {
-        flag: "/O2"
+        flag: "/O2" # Implies /Og /Oi /Ot /Oy /Ob2 /Gs /GF /Gy
         flag: "/DNDEBUG"
       }
     }
   }
 
+  feature {
+    name: 'treat_warnings_as_errors'
+    flag_set {
+      action: 'c-compile'
+      action: 'c++-compile'
+      flag_group {
+        flag: "/WX"
+      }
+    }
+  }
+
+  # Enable C++ exception by default.
+  feature {
+    name: 'default_exceptions'
+    enabled: true
+    flag_set {
+      action: 'c++-compile'
+      with_feature: {
+        not_feature: 'no_exceptions'
+        not_feature: 'use_exceptions'
+      }
+      flag_group {
+        flag: "/D_HAS_EXCEPTIONS=1"
+        flag: "/EHsc"
+      }
+    }
+  }
+
+  feature {
+    name: 'use_exceptions'
+    flag_set {
+      action: 'c++-compile'
+      with_feature: {
+        not_feature: 'no_exceptions'
+      }
+      flag_group {
+        flag: "/D_HAS_EXCEPTIONS=1"
+        flag: "/EHsc"
+      }
+    }
+  }
+
+  feature {
+    name: 'no_exceptions'
+    flag_set {
+      action: 'c++-compile'
+      with_feature: {
+        not_feature: 'use_exceptions'
+      }
+      flag_group {
+        flag: "/D_HAS_EXCEPTIONS=0"
+        flag: "/EHs-c-"
+        flag: "/wd4577" # Suppress 'noexcept used with no exception handling mode specified'
+      }
+    }
+  }
+
+  # 'default_rtti' omitted as RTTI is always enabled by default in MSVC
+
+  feature {
+    name: 'use_rtti'
+    flag_set {
+      action: 'c++-compile'
+      with_feature: {
+        not_feature: 'no_rtti'
+      }
+      flag_group {
+        flag: "/GR"
+      }
+    }
+  }
+
+  feature {
+    name: 'no_rtti'
+    flag_set {
+      action: 'c++-compile'
+      with_feature: {
+        not_feature: 'use_rtti'
+        not_feature: 'use_exceptions' # RTTI is needed for exceptions to work
```

Comment:
Can you elaborate here? I'd think that use_exceptions cannot be enabled without rtti, not the other way around? There's also requires: for strict checking

Reply:
> I'd think that use_exceptions cannot be enabled without rtti, not the other way around?

I was thinking in the opposite way. I think your way is better.

Reply:
I have decided to rollback `use_rtti` and `no_rtti` for now. `default_exceptions` makes it harder to make sure user do not enable `use_exceptions` and `no_rtti` together. Before there is proper documentation on how each feature should be used is written on https://docs.bazel.build, it is better not leaving this dangerous feature lying around like that.

Depending on compilers, it is generally unsafe to mix `-frtti` and `-fno-rtti` code in one binary as RTTI can affect ABI. `-fexceptions` and `fno-exceptions` in one code is usually OK as long as exceptions are not thrown across boundary.